setup/nextcloud.sh

```diff
@@ -328,11 +289,57 @@ hide_output sudo -u www-data php$PHP_VER /usr/local/lib/owncloud/console.php app
 sudo -u www-data php$PHP_VER /usr/local/lib/owncloud/occ upgrade
 if [ \( $? -ne 0 \) -a \( $? -ne 3 \) ]; then exit 1; fi
 
+# Turn off read only in case it wasn't turned off before.
+sed -ie '/config_is_read_only/d' $STORAGE_ROOT/owncloud/config.php
+
 # Disable default apps that we don't support
 sudo -u www-data \
-	php$PHP_VER /usr/local/lib/owncloud/occ app:disable photos dashboard activity \
+	php$PHP_VER /usr/local/lib/owncloud/occ app:disable \
+		photos dashboard activity circles federation files_sharing \
+		notifications files_pdfviewer password_policy systemtags comments \
+		privacy recommendations files_rightclick sharebymail support text \
+		theming survey_client user_status weather_status files_videoplayer \
+		contactsinteraction \
 	| (grep -v "No such app enabled" || /bin/true)
 
+# Update config.php.
+# * trusted_domains is reset to localhost by autoconfig starting with ownCloud 8.1.1,
+#   so set it here. It also can change if the box's PRIMARY_HOSTNAME changes, so
+#   this will make sure it has the right value.
+# * Some settings weren't included in previous versions of Mail-in-a-Box.
+# * We need to set the timezone to the system timezone to allow fail2ban to ban
+#   users within the proper timeframe
+# * We need to set the logdateformat to something that will work correctly with fail2ban
+# * mail_domain' needs to be set every time we run the setup. Making sure we are setting
+#   the correct domain name if the domain is being change from the previous setup.
+# Use PHP to read the settings file, modify it, and write out the new settings array.
+TIMEZONE=$(cat /etc/timezone)
+instanceid=oc$(echo $PRIMARY_HOSTNAME | sha1sum | fold -w 10 | head -n 1)
+CONFIG_TEMP=/tmp/cfg-$instanceid.json
+PHONE_REGION=$(locale | grep TELEPHONE | sed -e 's/LC_.*=".*_//' | sed -e 's/\..*//')
```

Comment:
For me this results in "C as the region. The " causes a problem on upgrading since it's invalid json. However C as a region isn't valid ISO 3166-1, which is the format NC requires: https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/config_sample_php_parameters.html?highlight=default_phone_region

I've removed it on my instance and left the default value.

Reply:
Without this, nextcloud was giving me a warning. For me, I am extracting 'US' from the locale.

I might need to find a more reliable way to get this code. I'll do some research on it.

Reply:
```locale
LANG=C.UTF-8
LANGUAGE=
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=```

Reply:
I have added to the code some checks if the locale is not set to something that provides a country code. In this case, it leaves the country code blank. This will cause a warning with nextcloud admin panel, but I don't think its too serious.